distribution/pull_v2_test.go

```diff
@@ -205,3 +215,190 @@ func TestFormatPlatform(t *testing.T) {
 		}
 	}
 }
+
+func Test_v2Puller_pullSchema2Config_Basic(t *testing.T) {
+	ctx := context.Background()
+
+	const imageJSON = `{
+	"architecture": "amd64",
+	"os": "linux",
+	"config": {},
+	"rootfs": {
+		"type": "layers",
+		"diff_ids": []
+	}
+}`
```

Comment:
I haven't checked out the branch locally yet to compare, but glancing over the tests, it looks like there's quite some common boilerplating code. Perhaps it's possible to combine (some of) the tests and use a sub-tests.

Just some quick examples that use sub-tests; 

https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/daemon/volumes_linux_test.go#L10

https://github.com/moby/moby/blob/9e8cf1016ead200aea31d7ff381ba2ed01416ddb/builder/dockerfile/evaluator_test.go#L27